decidim-core/app/controllers/decidim/newsletters_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Decidim
+  # The controller to show the newsletter on the website.
+  class NewslettersController < Decidim::ApplicationController
+    skip_authorization_check
+
+    layout "decidim/mailer", only: [:show]
+    helper Decidim::SanitizeHelper
+    include Decidim::NewslettersHelper
+
+    helper_method :newsletter
+
+    def show
+      @user = current_user
+      @organization = current_organization
+
+      if newsletter.sent?
+        if @user.present?
+          @string_encrypted = Decidim::Encryptor.sent_at_encrypted(@user.id, newsletter.sent_at)
```

Comment:
I would use `@encrypted_token` for the variable name instead as it's more self-explanative.

Reply:
Also, it looks like the test don't cover this scenario, which is quite critical.